gems/smithy/lib/smithy/vise/structure_parser.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Vise
+    # @api private
+    class StructureParser
+      PRELUDE_SHAPES = %w[
+        smithy.api#BigInteger
+        smithy.api#BigDecimal
+        smithy.api#Blob
+        smithy.api#Boolean
+        smithy.api#Byte
+        smithy.api#Document
+        smithy.api#Double
+        smithy.api#Float
+        smithy.api#Integer
+        smithy.api#Long
+        smithy.api#PrimitiveBoolean
+        smithy.api#PrimitiveByte
+        smithy.api#PrimitiveDouble
+        smithy.api#PrimitiveFloat
+        smithy.api#PrimitiveInteger
+        smithy.api#PrimitiveLong
+        smithy.api#PrimitiveShort
+        smithy.api#Short
+        smithy.api#String
+        smithy.api#Timestamp
+        smithy.api#Unit
+      ].freeze
+
+      def initialize(model)
+        @shapes = model['shapes']
+      end
+
+      def shapes_for(structure)
+        shapes = {}
+        _id, structure = structure.first
+        structure['members']&.each_value { |member| parse_member(member, shapes) }
+        shapes
+      end
+
+      private
+
+      def parse_member(member_ref, shapes)
+        target = member_ref['target']
+        return if PRELUDE_SHAPES.include?(target)
```

Comment:
I'm not quite sure this is how we should be handeling prelude shapes.  This leads to code that uses the model also needing to know about prelude shapes.

I'd potentially lean towards adding a default weld with a pre-process that checks the model for prelude shapes, and if missing, adds them in.  That ensures model integrity and simplifies view code.

Reply:
Do prelude shapes get included in c2j conversions of a smithy model?

Reply:
Without doing a deep dive into the code - the answer is yes, but they get translated (possibly a synthetic shape is created?). Here's an example:
* Smithy model: https://github.com/aws/aws-models/blob/master/notifications/smithy/model.json#L2145
* C2J: https://github.com/aws/aws-models/blob/master/notifications/2018-05-10/api-2.json#L1104

Reply:
Hm ok. It's slightly different since c2j has type fields. We probably want to add prelude shapes to the model in a preprocess only if they are referenced. Quite possibly we should do it before preprocess hooks too (bake it in). The prelude option in smithy run I don't believe adds these (can you check?)

Reply:
Yeah - I think ideally we would add them only if they are referenced, but its simpler to just add them all and (not including traits), there aren't many and we have code that gets the service closure anyway (ie, this pr).  

Just tried with smithy run with the model plugin with `includePreludeShapes: true` - it does not add the "synthetic" like in c2j, just adds definitions for the shapes like:
```
"smithy.api#String": {
            "type": "string"
        },
```

It does also add all of the trait definitions too which we don't need.

Reply:
I think the code still needs to check for prelude shapes by checking for a types key. We can possibly go without changing the model. Let's see how this goes as we add other code generation pieces.

Reply:
How would it be handled for documentation? Some logic would need to check the existence of the type key and have a map of type to ruby type. But we may as well have a map of prelude shapes id to Ruby type.

Reply:
For documentation its much simpler if you have the prelude types in the model.  I think it reduces special casing.  Consider the following model:

```json
{
  "example#S": {
      "type": "structure",
      "members": {
          "member1": {
              "target": "example#NonPrelude"
          },
          "member2": {
              "target": "smithy.api#String"
          }
      }
  },
  "example#NonPrelude": {
            "type": "string"
  }
}
```

Without having the prelude String defined, you need a check on target before getting it to get the smithy type. But if its included in the model, then you just get the member's target and get the type from it.  Then map the Smithy type to Ruby Type.  No special case check of prelude shape id first.

Reply:
Possibly - I think in either case we need a mapping of something to ruby type, and we can map prelude shape target to ruby class and avoid adding to the model and indexing into members an extra step. We can add this in another PR.